File: ios/RNModule/RNModule/RNTableViewListener.m
//
//  RNTableViewListener.m
//  RNModule
//
//  Created by 홍창남 on 13/11/2018.
//  Copyright © 2018 홍창남. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTConvert.h>

@interface RCT_EXTERN_MODULE(RNTableViewListener, NSObject)

RCT_EXTERN_METHOD(didSelectTableViewCell:(NSInteger *)sectionIndex rowIndex:(NSInteger *)rowIndex)

RCT_EXTERN_METHOD(didSelectSectionListCell:(NSDictionary *)userInfo sectionIndex:(NSInteger *)sectionIndex rowIndex:(NSInteger *)rowIndex)

@end
//RCT_EXTERN_METHOD(getCount:(NSString)name withAge:(NSNumber)age)
//func getCount(_ name: NSString, withAge age: NSNumber) {...}

//func decrement(
//               _ resolve: RCTPromiseResolveBlock,
//               rejecter reject: RCTPromiseRejectBlock
//               ) -> Void {
//RCT_EXTERN_METHOD(
//                  decrement: (RCTPromiseResolveBlock)resolve
//                  rejecter: (RCTPromiseRejectBlock)reject
//                  )
//CalendarManager.addEvent('Birthday Party', {
//location: '4 Privet Drive, Surrey',
//time: date.getTime(),
//description: '...',
//});

//@objc func download(_ fileUrl: String, callback: RCTResponseSenderBlock) -> Void
//
//# Bridging module
//RCT_EXTERN_METHOD(download:(NSString*)fileUrl callback:(RCTResponseSenderBlock))

File: ios/RNModule/RNModule/RNTableViewListener.swift
//
//  RNTableViewListener.swift
//  RNModule
//
//  Created by 홍창남 on 13/11/2018.
//  Copyright © 2018 홍창남. All rights reserved.
//

import Foundation
import React

@objc(RNTableViewListener)
class RNTableViewListener: NSObject {
    
    @objc static func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    @objc func didSelectTableViewCell(_ sectionIndex: Int, rowIndex: Int) {
        
        let indexPath = IndexPath(row: rowIndex, section: sectionIndex)
        NotificationCenter.default.post(name: NSNotification.Name(rawValue: "didSelectTableViewCell"),
                                        object: nil, userInfo: ["indexPath": indexPath])
    }
    
    @objc func didSelectSectionListCell(_ userInfo: Dictionary<AnyHashable, Any>, sectionIndex: Int, rowIndex: Int) {
        
        let indexPath = IndexPath(row: rowIndex, section: sectionIndex)
        print(indexPath)
        print(userInfo)
    }
}
//func getCount(_ name: NSString, withAge age: NSNumber) {...}

